app/src/main/java/org/jellyfin/androidtv/TvApp.java

```diff
@@ -237,27 +227,7 @@ public void setPlaybackController(PlaybackController playbackController) {
     }
 
     public void showSearch(final Activity activity, boolean musicOnly) {
-        if (!voiceSearchAllowed &&
-                ContextCompat.checkSelfPermission(activity, Manifest.permission.RECORD_AUDIO) != PackageManager.PERMISSION_GRANTED) {
-            //request necessary permission
-            logger.Info("Requesting search permission...");
-            if (ActivityCompat.shouldShowRequestPermissionRationale(activity, Manifest.permission.RECORD_AUDIO)) {
-                //show explanation
-                logger.Info("Show rationale for permission");
-                new AlertDialog.Builder(activity)
-                        .setTitle("Search Permission")
-                        .setMessage("Search requires permission to record audio in order to use the microphone for voice search")
-                        .setPositiveButton("Ok", new DialogInterface.OnClickListener() {
-                            public void onClick(DialogInterface dialog, int whichButton) {
-                                ActivityCompat.requestPermissions(activity, new String[] {Manifest.permission.RECORD_AUDIO}, SEARCH_PERMISSION);
-                            }
-                        }).show();
-            } else {
-                ActivityCompat.requestPermissions(activity, new String[] {Manifest.permission.RECORD_AUDIO}, SEARCH_PERMISSION);
-            }
-        } else {
             showSearchWithPermission(activity, musicOnly);
```

Comment:
Perhaps we should remove the `showSearchWithPermission` method and just move the implementation here. It no longer seems necessary since the permissions aren’t handled here now.

Reply:
Good one!